### ISSUE TITLE
[Android] CameraUI fail with App Bundle

### ISSUE THREAD

**Issue by esdebon** [airsdk/Adobe-Runtime-Support#2666]
Using CameraUI to take a picture, I noticed that it only works by creating the APK in the old way (without the Android App Bundle) this with the targetSdkVersion less than 22.

When using a higher version (with Android App Bundle), the camera tries to create the file but the file size is 0 kb, i.e. only empty files are generated.

Reviewing the AndroidManifest.xml file generated with the KeepAndroidStudioOutput flag set, I noticed that in the references for the writing of the files generated by the camera, the prefix air is added to the App Id

This happens on Android with SDK from 33.1.1.743 to 50.2.2.5 and I have tested it on a Samsung Galaxy S9 running Android 10 and on a Galaxy S20.  

![Captura de pantalla (828)](https://github.com/airsdk/Adobe-Runtime-Support/assets/1408296/cb2f471b-0815-4dec-a469-b3e01af1dfd1)

**Comment by esdebon** [comment]
With SDK 50.2.2.6 the error persists

air.air.com.domainTest.app.provider instead air.com.domainTest.app.provider in the AndroidManifest.xml auto generated

```
<provider
            android:name="com.adobe.air.CameraUIProvider"
        android:authorities="air.air.com.domainTest.app.provider"
        android:exported="false"
        android:grantUriPermissions="true" >
        <meta-data
            android:name="android.support.FILE_PROVIDER_PATHS"
            android:resource="@xml/provider_paths" >
        </meta-data>
    </provider>
```

**Comment by marchbold** [comment]
What is your application id in your app descriptor?

**Comment by esdebon** [comment]
```
<application xmlns="http://ns.adobe.com/air/application/50.2">
  <id>air.com.domainTest.app.provider</id>
  <versionNumber>1.0.0</versionNumber>
  <versionLabel/>

```
But in the file adt.cfg the flag AddAirToAppID is false

`AddAirToAppID=false`

**Comment by marchbold** [comment]
Are you using apm to generate your app descriptor? Can you tell me which `adt.cfg` you have edited?

`apm` can only pick up certain configs:

- values set in the user adt.cfg (`~/.airsdk/adt.cfg`) This is what is set via the air sdk manager
- `AIR_NOANDROIDFLAIR` set on the terminal (apm must be run in the same environment as your build)
- `adt.cfg` in the `AIR_HOME` directory if you have set that environment variable
- `adt.cfg` in the AIR SDK used to run `apm`, ie. what ever you have set as your default `adl` 

If you run `apm` in a different environment from your build then it can't detect the `adt.cfg` settings there.

**Comment by esdebon** [comment]
I have done the generation of the AAB file in two ways and obtained the same result.

1.- From Adobe Animate:

- Adding the SDK from Help>Manage Adobe AIR SDK.
- Adding the environment variable AIR_NOANDROIDFLAIR = true from the windows control panel

2.- From the windows command line with the following script:

```
@ECHO OFF

setlocal
title Compiling AAB
cls
echo ====================================================
echo "COMPILING APP"
echo ====================================================

SET AIR_NOANDROIDFLAIR=true

SET PROJECT=App
SET AIRPATH=C:\Users\xxxxxxxx\Documents\AIRSDK\AIR50.2.2.6
SET ADT="%AIRPATH%\bin\adt"

SET NAME=Main

SET CERT_PSWD=XXXXXXXXXXXXXX
SET SIGNING_OPTIONS=-storetype pkcs12 -storepass %CERT_PSWD% -keystore "../Android2.p12"

SET EXTDIR="../lib"

SET ASSETS="./_textures" "./_fonts" "./_particles" "./_audio" "./AppIconsForPublish" "./loadingScreen.png" "./lang.xml" 

%ADT% -package -target aab %SIGNING_OPTIONS% "%PROJECT%.aab" "%NAME%-app.xml" "Main.swf" %ASSETS% -extdir %EXTDIR% 
```

In both cases before compiling I delete the AndroidStudioProject folder and in both cases a new one is created with the same problem of adding the prefix "air" to the app id in the AndroidManifest.xml file.

**Comment by ajwfrost** [comment]
Interesting -> so this appears to be a bug in ADT, where there is always a `air.` prefix being added to the camera UI provider name, regardless of whether that configuration means the prefix is applied to the Android package name. Internal inconsistency in how this is used...

It's a quick fix so we can get this in to the forthcoming release. In the meantime I'd have thought you could remove the `air.` from your actual app descriptor file and then change the config so that it gets added back everywhere..?

**Comment by marchbold** [comment]
(oh I had thought this was from an extension but this is the built in AIR UI, so yeah ignore all my comments about `apm` as that's unrelated).

**Comment by esdebon** [comment]
Fixed in the SDK 50.2.3.1

![issue2666](https://github.com/airsdk/Adobe-Runtime-Support/assets/1408296/e070c820-18a6-458c-b848-0c4d521cf511)
